src/python_testing/TC_EWATERHTRM_1_2.py

```diff
@@ -0,0 +1,158 @@
+# === END CI TEST ARGUMENTS ===
+
+import logging
+
+import chip.clusters as Clusters
+from matter_testing_support import MatterBaseTest, TestStep, async_test_body, default_matter_test_main
+from mobly import asserts
+
+
+class TC_EWATERHTRM_1_2(MatterBaseTest):
+
+    async def read_mode_attribute_expect_success(self, endpoint, attribute):
+        cluster = Clusters.Objects.WaterHeaterMode
+        return await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=attribute)
+
+    def desc_TC_EWATERHTRM_1_2(self) -> str:
+        return "[TC-EWATERHTRM-1.2] Cluster attributes with DUT as Server"
+
+    def steps_TC_EWATERHTRM_1_2(self) -> list[TestStep]:
+        steps = [
+            TestStep(1, "Commissioning, already done", is_commissioning=True),
+            TestStep(2, "Read the SupportedModes attribute"),
+            TestStep(3, "Read the CurrentMode attribute"),
+        ]
+        return steps
+
+    def pics_TC_EWATERHTRM_1_2(self) -> list[str]:
+        pics = [
+            "EWATERHTRM.S",
+        ]
+        return pics
+
+    @async_test_body
+    async def test_TC_EWATERHTRM_1_2(self):
+
+        endpoint = self.user_params.get("endpoint", 1)
+
+        attributes = Clusters.WaterHeaterMode.Attributes
+
+        self.step(1)
+
+        self.step(2)
+        supported_modes = await self.read_mode_attribute_expect_success(endpoint=endpoint, attribute=attributes.SupportedModes)
+        asserts.assert_greater_equal(len(supported_modes), 2,
+                                     "SupportedModes must have at least 2 entries!")
+        asserts.assert_less_equal(len(supported_modes), 255,
+                                  "SupportedModes must have at most 255 entries!")
+        modes = set([m.mode for m in supported_modes])
+        asserts.assert_equal(len(modes), len(supported_modes),
+                             "SupportedModes must have unique mode values")
+
+        labels = set([m.label for m in supported_modes])
+        asserts.assert_equal(len(labels), len(supported_modes),
+                             "SupportedModes must have unique mode label values")
+
+        # common mode tags
+        commonTags = {0x0: 'Auto',
+                      0x1: 'Quick',
+                      0x2: 'Quiet',
+                      0x3: 'LowNoise',
+                      0x4: 'LowEnergy',
+                      0x5: 'Vacation',
+                      0x6: 'Min',
+                      0x7: 'Max',
+                      0x8: 'Night',
+                      0x9: 'Day'}
+
+        # derived cluster defined tags
+        # kUnknownEnumValue may not be defined
+        try:
+            derivedTags = [tag.value for tag in Clusters.WaterHeaterMode.Enums.ModeTag
+                           if tag is not Clusters.WaterHeaterMode.Enums.ModeTag.kUnknownEnumValue]
+        except AttributeError:
+            derivedTags = Clusters.WaterHeaterMode.Enums.ModeTag
```

Comment:
could you add a comment on why AttributeError path may be take and is acceptable? the comment regarding kUnknown only explains why we skip that one on list conversion.

Reply:
I have removed the need for this try/catch